docs/continuous_integration.rst

```diff
@@ -0,0 +1,60 @@
+.. _ci_tests:
+
+Continuous Integration
+===============================
+
+The SecureDrop project uses `GitHub Actions <https://docs.github.com/en/actions>`_ for
+running automated continuous integration on code changes. You can get an overview of what
+each project does by reviewing the ``Makefile`` and files in the ``.github/workflows`` folder.
+
+Basics
+------
+
+Our CI runs a mixture of linters, tests and build processes to validate code submissions.
+
+Most tasks have a corresponding ``make`` target that will run the same command locally. Common
+targets across all our projects include:
+
+* ``make lint``: run linters
+* ``make fix``: apply automated fixes from formatters and linters
+* ``make test``: run automated tests
+
+In CI, these are run in a container using the corresponding Linux distribution (e.g. Debian or Fedora),
+which can also be used to reproduce CI results locally. Some projects, like ``securedrop`` (server) and
+``securedrop-workstation``, automatically run commands in containers.
+
+Pull requests
+-------------
+
+Most CI jobs are triggered by both ``push`` and ``pull_request`` events. The former is run against
+your branch, while the latter is run against your branch merged into ``main`` (or ``develop``).
+
+
+Special branch prefixes
+-----------------------
+
+In the ``securedrop`` repository, some slower jobs are only triggered if a specific branch prefix
+is used when creating the pull request. Currently these are:
+
+* ``stg-*``: runs a staging build in GCE, see :ref:`Configuration Tests<config_tests>`
+* ``l10n-*``: runs localization tests across all 20+ supported languages
+
+Nightlies
+---------
+
+For ``securedrop-workstation`` and ``securedrop-client``, we build packages on every merged
+commit and every night and publish them to our test yum and apt repositories respectively.
```

Comment:
Is it worth specifying when "every night" is? Not sure if that's the kind of detail worth putting in the docs or if people want to know they can go lookup the workflow.

Reply:
Yeah, I think if people want to know the exact time they can just look it up.